presto-main/src/main/java/com/facebook/presto/sql/rewrite/ShowQueriesRewrite.java

```diff
@@ -215,7 +215,14 @@ protected Node visitShowTables(ShowTables showTables, Void context)
 
             Optional<String> likePattern = showTables.getLikePattern();
             if (likePattern.isPresent()) {
-                Expression likePredicate = new LikePredicate(identifier("table_name"), new StringLiteral(likePattern.get()), null);
+                Expression likePredicate;
+                Optional<String> escape = showTables.getEscape();
+                if (escape.isPresent()) {
+                    likePredicate = new LikePredicate(identifier("table_name"), new StringLiteral(likePattern.get()), new StringLiteral(escape.get()));
+                }
+                else {
+                    likePredicate = new LikePredicate(identifier("table_name"), new StringLiteral(likePattern.get()), null);
```

Comment:
Please add a constructor `LikePredicate(Expression value, Expression pattern, Optional<Expression> escape)`, we should avoid `@Nullable` parameter in the constructor.

Reply:
I find LikePredicate used in somewhere else, and LikePredicate  is used with null parameter. 
Eg:
presto-main/src/main/java/com/facebook/presto/sql/planner/ExpressionInterpreter.java
line 1093~1098:
            Expression optimizedEscape = null;
            if (node.getEscape() != null) {
                optimizedEscape = toExpression(escape, type(node.getEscape()));
            }
            return new LikePredicate(
                    toExpression(value, type(node.getValue())),
                    toExpression(pattern, type(node.getPattern())),
                    optimizedEscape);

I change all of them?